internal/resources/akscluster/resource_nodepool.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package akscluster
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client"
+	aksnodepool "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/akscluster/nodepool"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	aksmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/akscluster"
+)
+
+// createNodepools sends the create request for the given nodepools as part of cluster creation flow.
+func createNodepools(ctx context.Context, nodepools []*aksmodel.VmwareTanzuManageV1alpha1AksclusterNodepoolNodepool, client aksnodepool.ClientService) error {
+	for _, np := range nodepools {
+		if err := createNodepool(ctx, np, client); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// handleNodepoolChanges if nodepool changes are detected delegates to the appropriate node pool operation: `Create`, `Update`, `Delete`.
+func handleNodepoolChanges(ctx context.Context, existing []*aksmodel.VmwareTanzuManageV1alpha1AksclusterNodepoolNodepool, data *schema.ResourceData, tc *client.TanzuMissionControl) error {
+	var created, updated, deleted []*aksmodel.VmwareTanzuManageV1alpha1AksclusterNodepoolNodepool
+
+	cfn := extractClusterFullName(data)
+	npCount := changedNodeCount(data)
+
+	for i := 0; i <= npCount; i++ {
+		key := fmt.Sprintf("spec.0.nodepool.%v", i)
+		if data.HasChange(key) {
+			oldResource, newResource := data.GetChange(key)
+
+			// The existence of an old and new state signals and update operation.
+			if !isEmpty(oldResource) && !isEmpty(newResource) {
+				np := constructNodepool(cfn, newResource.(map[string]any))
+				updated = append(updated, np)
+
+				continue
+			}
+
+			// If the old resource is empty this signals a new nodepool has been created.
+			if isEmpty(oldResource) {
+				np := constructNodepool(cfn, newResource.(map[string]any))
+				created = append(created, np)
+
+				continue
+			}
+
+			// If the new version is empty this signals a node pool has been deleted.
+			if isEmpty(newResource) {
+				np := constructNodepool(cfn, oldResource.(map[string]any))
+				deleted = append(deleted, np)
+
+				continue
+			}
+		}
+	}
+
+	npData := map[string][]*aksmodel.VmwareTanzuManageV1alpha1AksclusterNodepoolNodepool{
```

Comment:
Nit: this is structured info that is passed into another method. Can we make a struct out of it or pass them separately?